python_ta/contracts/__init__.py

```diff
@@ -176,18 +176,21 @@ def new_setattr(self: klass, name: str, value: Any) -> None:
                     f"Value {_display_value(value)} did not match type annotation for attribute "
                     f"{name}: {_display_annotation(cls_annotations[name])}"
                 ) from None
-
+        original_attr_value = None
+        if hasattr(super(klass, self), name):
+            original_attr_value = super(klass, self).__getattribute__(name)
         super(klass, self).__setattr__(name, value)
-        curframe = inspect.currentframe()
-        callframe = inspect.getouterframes(curframe, 2)
-        frame_locals = callframe[1].frame.f_locals
+        frame_locals = inspect.currentframe().f_back.f_locals
         if self is not frame_locals.get("self"):
             # Only validating if the attribute is not being set in a instance/class method
-            klass_mod = _get_module(klass)
             if klass_mod is not None and ENABLE_CONTRACT_CHECKING:
                 try:
                     _check_invariants(self, klass, klass_mod.__dict__)
                 except PyTAContractError as e:
+                    if original_attr_value is None:
```

Comment:
This check doesn't work if the original attribute value was `None`, which is a different case from the original attribute note existing at all. You can use a separate variable to track whether there was an original attribute. And make sure to add a test for this case as well.